src/bindings/c/tests/ov_auto_property_test.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ov_test.hpp"
+
+using ConfigParams = std::tuple<std::string, const char*, const char*>;
+
+class ov_auto_plugin_test : public ::testing::TestWithParam<ConfigParams> {
+public:
+    std::string device_name;
+    const char* auto_property;
+    const char* property_value;
+
+public:
+    void SetUp() override {
+        std::tie(device_name, auto_property, property_value) = GetParam();
+    }
+};
+
+TEST_P(ov_auto_plugin_test, ov_core_auto_set_and_get_property_bool) {
+    ov_core_t* core = nullptr;
+    OV_EXPECT_OK(ov_core_create(&core));
+    EXPECT_NE(nullptr, core);
+    OV_EXPECT_OK(ov_core_set_property(core, device_name.c_str(), auto_property, property_value));
+    char* ret = nullptr;
+    OV_EXPECT_OK(ov_core_get_property(core, device_name.c_str(), auto_property, &ret));
+    EXPECT_STREQ(property_value, ret);
+    ov_free(ret);
```

Comment:
Please add test case for invalid input value.

Reply:
Added.